templates/github/.github/workflows/release.yml.j2

```diff
@@ -127,30 +176,10 @@ jobs:
         env:
           SECRETS_CONTEXT: {{ "${{ toJson(secrets) }}" }}
 
-      {%- if deploy_to_pypi %}
-      - name: Deploy plugin to pypi
-        run: bash .ci/scripts/publish_plugin_pypi.sh
-      {%- endif %}
-
-      {%- if deploy_client_to_rubygems %}
-      - name: Publish client to rubygems
-        run: bash .ci/scripts/publish_client_gem.sh
-      {%- endif %}
-
-      {%- if deploy_client_to_pypi %}
-      - name: Publish client to pypi
-        run: bash .ci/scripts/publish_client_pypi.sh
-      {%- endif %}
-
-      {%- if publish_docs_to_pulpprojectdotorg %}
-      - name: Publish docs to pulpproject.org
-        run: .ci/scripts/publish_docs.sh tag ${GITHUB_REF##*/}
-      {%- endif %}
-
-      {%- if update_redmine %}
-      - name: Update Redmine
-        run: bash .ci/scripts/update_redmine.sh
-      {%- endif %}
+      - name: {{ "${{ matrix.name }}" }}
+        run: |
+          git checkout "origin/${GITHUB_REF##*/}" -- .ci
```

Comment:
It’s probably better to use the branch’s ci scripts rather than the tag’s version of the ci scripts.

Reply:
I'm not quite following here. Why?
And which branch do you mean? "2.2" for a "2.2.5" release?

Reply:
This workflow runs against the tag (see [here](https://github.com/pulp/plugin_template/pull/357/files#diff-2804d6731049ff36e9d9c9de50d1541440740aa9fd2479a75714f4e1bfc361ddR132)). It does so in order to build and publish the tag's files to pypi, docs.pulpproject.org, etc.

However, consider the case that you try to release a tag but encounter a problem in the CI scripts. You fix the CI scripts but the workflow will still run against the tag's version of the CI scripts. So the only fix would be to create a new tag with new ci files.

To prevent such a problem, I thought we could check out the .ci scripts from the branch instead.

Reply:
Maybe I am overthinking things though?

Reply:
So in case the Release failed, and rerunning does not suffice, but you need to fix CI files, you push them to the release branch and then restart? Maybe this _is_ solving non-obvious corner-case problems. But we should have this troubleshooting method documented somewhere and linked from/to this comment.

Reply:
I added a comment to the release workflow. I'll also add a section about troubleshooting failed releases to our release guide.